packages/tocco-ui/src/Panel/PanelGroup.js

```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import StyledPanelGroup from './StyledPanelGroup'
+
+class PanelGroup extends React.Component {
+  state = {
+    openPanelId: this.props.openPanelId >= 0 ? this.props.openPanelId : -1
+  }
+
+  toggleOpenStateAsGroup = panelId => {
+    this.setState({openPanelId: panelId})
+  }
+
+  render() {
+    return (
+      <StyledPanelGroup>
+        {
+          React.Children.map(this.props.children, (child, i) =>
+            React.cloneElement(child, {
+              isOpen: this.state.openPanelId === i,
+              panelId: i,
+              toggleOpenStateAsGroup: this.toggleOpenStateAsGroup
```

Comment:
Actually, the child doesn't have to receive a `panelId`, you can simply use the index without passing it down to the child.

what about doing it like this:

```
handleToggle = (index, open) => {
  this.setState({
    openPanelIndex: open === true ? index : undefined
  })
}

render () {
  return (
    <StyledPanelGroup>
      {
        React.Children.map(this.props.children, (child, i) =>
          React.cloneElement(child, {
            isOpen: this.state.openPanelIndex === i,
            onToggle: this.handleToggle.bind(this, i)
          })
        )
     }
    </StyledPanelGroup>
  )
}